### ISSUE TITLE
Images unable to build with docker "BuildKit"

### ISSUE THREAD

**Issue by rocketeerbkw** [uselagoon/lagoon#2297]
# Describe the bug

A comedy of errors:

1) Some images (so far we know `harborregistry` and `harborregistryctl`) cannot be built with the [BuildKit](https://docs.docker.com/develop/develop-images/build_enhancements/) docker build engine.
2) BuildKit is enabled by default now in Docker for Mac [2.4.0.0](https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-community-2400).
3) Installing Docker for Mac fresh or resetting to factory defaults cause `make build` to fail.

## To Reproduce

1. Enable BuildKit
2. Try to build lagoon images

## Expected behavior

Users can build lagoon images no matter what build engine is enabled

## Screenshots

n/a

## Additional context

The error with `harborregistry`:

```
docker build --no-cache --build-arg LAGOON_VERSION=v1.10.0 --build-arg IMAGE_REPO=lagoon --build-arg UPSTREAM_REPO=uselagoon --build-arg UPSTREAM_TAG=latest --build-arg ALPINE_VERSION=3.11 -t lagoon/harborregistry -f services/harborregistry/Dockerfile services/harborregistry
[+] Building 0.7s (8/8) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                      0.0s
 => => transferring dockerfile: 37B                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                         0.0s
 => => transferring context: 2B                                                                                                                           0.0s
 => [internal] load metadata for docker.io/goharbor/registry-photon:v2.7.1-patch-2819-2553-v1.10.0                                                        0.4s
 => [internal] load build context                                                                                                                         0.0s
 => => transferring context: 70B                                                                                                                          0.0s
 => CACHED [1/4] FROM docker.io/goharbor/registry-photon:v2.7.1-patch-2819-2553-v1.10.0@sha256:873c73954aaae94267cecdeee83676c04a03dbb3230473ba478ec94fa  0.0s
 => [2/4] COPY entrypoint.sh /home/harbor/                                                                                                                0.0s
 => [3/4] COPY --chown=harbor:root install_cert.sh /home/harbor/                                                                                          0.0s
 => ERROR [4/4] RUN chmod -R g+w /etc/pki/tls/certs && chgrp -R root /etc/pki/tls/certs && chmod -R g+w /var/lib/registry && chmod -R g+w /etc/registry   0.2s
------
 > [4/4] RUN chmod -R g+w /etc/pki/tls/certs && chgrp -R root /etc/pki/tls/certs && chmod -R g+w /var/lib/registry && chmod -R g+w /etc/registry && chgrp -R root /home/harbor && chmod -R g+w /home/harbor && if [ -f /storage ]; then chmod -R g+w /storage; fi:
#8 0.171 chmod: /var/lib/registry: No such file or directory
------
failed to solve with frontend dockerfile.v0: failed to build LLB: executor failed running [/bin/sh -c chmod -R g+w /etc/pki/tls/certs && chgrp -R root /etc/pki/tls/certs && chmod -R g+w /var/lib/registry && chmod -R g+w /etc/registry && chgrp -R root /home/harbor && chmod -R g+w /home/harbor && if [ -f /storage ]; then chmod -R g+w /storage; fi]: runc did not terminate sucessfully
make: *** [build/harborregistry] Error 1
```

**Comment by rocketeerbkw** [comment]
Not sure why the build fails, but the docs to state there is a known [limitation](https://docs.docker.com/develop/develop-images/build_enhancements/#limitations) that it's `Only supported for building Linux containers`.

Easiest fix now is to just update our documentation to warn people to [disable BuildKit](https://docs.docker.com/develop/develop-images/build_enhancements/#to-enable-buildkit-builds), but on the off chance that a user depends on that, it's not ideal that we require them to disable it just to run lagoon.

Should we open/search for upstream issues so that the broken images can work with buildkit?

We might be able to disable BuildKit by setting the `DOCKER_BUILDKIT` environment variable in our Makefile: https://docs.docker.com/develop/develop-images/build_enhancements/#to-enable-buildkit-builds.

**Comment by tobybellwood** [comment]
fixed in uselagoon/build-deploy-tool#143
